MappingThread.py

```diff
@@ -62,39 +76,32 @@ def plotMap(self, latitude, longitude):
         lon2 = longitude + radius / 111.320 / math.cos(lat2 * math.pi / 180.0)
         p2 = MapBox.MapPoint(lat2, lon2)  # Map corner 2
 
-        # Create MapPoints that correspond to corners of a square area (of side length 2*radius) surrounding the
-        # inputted latitude and longitude.
-
-        location = MapBox.TileGrid(p1, p2, zoom)
-        location.downloadArrayImages()
-
-        largeMapImage = location.genStichedMap()
-
         desiredSize = self.getDesiredMapSize()
 
-        if desiredSize is not None:
-            # Scale "to fit", maintains map aspect ratio even if it differs from the desired dimensions aspect ratio
-            scaleFactor = min(desiredSize[0] / largeMapImage.shape[0], desiredSize[1] / largeMapImage.shape[1])
-        else:
-            scaleFactor = 1
+        self.requestQueue.put_nowait((p1, p2, zoom, desiredSize))
 
-        scaleFactor = min(scaleFactor, 1) # You shall not scale the map larger. Waste of memory.
+        # NOTE: Passing numpy arrays through any sort of IPC other than plane old shared memory will result in the data
+        # being "pickled" and un-"pickled". This uses more processing time than just using shared memory. Since we're
+        # not worried about sync safety here, shared memory might be faster... but also harder to implement.
```

Comment:
Is this something to address in the future then?

Reply:
It's not an issue as it is right now. The numpy array is already scaled to the right size so its not that big. Also shared memory is a fixed size and this numpy array is a variable size so it would add a lot of complications to make that work. I'm not sure it's worth it.

If we can determine that we are still having performance issues and it's because of this, then we can change it for sure.